beacon-chain/blockchain/process_block.go

```diff
@@ -537,6 +537,27 @@ func (s *Service) InsertSlashingsToForkChoiceStore(ctx context.Context, slashing
 	}
 }
 
+func getBlockPayloadHash(blk interfaces.BeaconBlock) ([32]byte, error) {
+	var blockHashFromPayload [32]byte
+	if blocks.IsPreBellatrixVersion(blk.Version()) {
+		return blockHashFromPayload, nil
+	}
+	payload, err := blk.Body().ExecutionPayload()
+	switch {
+	case errors.Is(err, wrapper.ErrUnsupportedField):
+		payloadHeader, err := blk.Body().ExecutionPayloadHeader()
```

Comment:
Why not do this on the block type? A little odd to do this on error return and my worry is if `ErrUnsupportedField` changes this will silently fail